src/scrapers/isracard.ts

```diff
@@ -0,0 +1,234 @@
+          accountNumber: cardCharge.cardNumber,
+          processedDate: moment(
+            cardCharge.billingDate,
+            'DD/MM/YYYY'
+          ).toISOString(),
+        };
+      }
+    );
+
+    /* get transactions data */
+    const month = monthMoment.month() + 1;
+    const monthStr = month < 10 ? `0${month}` : month.toString();
+    const transUrl = `${SERVICE_URL}?reqName=CardsTransactionsList&month=${monthStr}&year=${monthMoment.year()}&requiredDate=N`;
+    const transResult = await fetchGetWithinPage<IsracardCardsTransactionsList>(
+      page,
+      transUrl
+    );
+
+    if (options && options.validateSchema) {
+      let validation = await validateSchema(
+        isracardCardsTransactionsList,
+        transResult
+      );
+      Object.assign(transResult, validation);
+    }
+
+    const accountTxns: { [key: string]: any } = {};
+    accounts.forEach((account) => {
+      const txnGroups = lodash.get(
+        transResult,
+        `CardsTransactionsListBean.Index${account.index}.CurrentCardTransactions`
+      );
+
+      if (txnGroups) {
+        let txnIsrael: any[] = [];
+        let txnAbroad: any[] = [];
+        txnGroups.forEach((txnGroup: { txnIsrael: any; txnAbroad: any }) => {
+          if (txnGroup.txnIsrael) {
+            const txns = txnGroup.txnIsrael.map((txn: any) => {
+              return Object.assign(txn, {
+                processedDate: account.processedDate,
+              });
+            });
+            txnIsrael.push(...txns);
+          }
+
+          if (txnGroup.txnAbroad) {
+            const txns = txnGroup.txnAbroad.map((txn: any) => {
+              return Object.assign(txn, {
+                processedDate: account.processedDate,
+              });
+            });
+            txnAbroad.push(...txns);
+          }
+        });
+        const accountNum = lodash.get(account, 'accountNumber');
+        accountTxns[accountNum] = {
+          accountNumber: accountNum,
+          index: account.index,
+          txnIsrael: txnIsrael,
+          txnAbroad: txnAbroad,
+        };
+      }
+    });
+    return accountTxns;
+  }
+  return 0;
+}
+
+async function getMonthDashboard(
+  page: puppeteer.Page,
+  monthMoment: moment.Moment,
+  options?: isracardOptions
+) {
+  // get accounts data
+  const billingDate = monthMoment.format('YYYY-MM-DD');
+  const accountsUrl = `${SERVICE_URL}?reqName=DashboardMonth&actionCode=0&billingDate=${billingDate}&format=Json`;
+  const getDashboardFunction = fetchGetWithinPage<IsracardDashboardMonth>(
+    page,
+    accountsUrl
+  );
+
+  if (options && options.validateSchema) {
+    const data = await getDashboardFunction;
+    let validation = await validateSchema(isracardDashboardMonth, data);
+    return {
+      data,
+      ...validation,
+    };
+  } else {
+    return { data: await getDashboardFunction };
+  }
+}
+
+async function getMonthTransactions(
+  page: puppeteer.Page,
+  monthMoment: moment.Moment,
```

Comment:
I think now that we expose it publicly, it's better to change this parameter to a regular date and do the Moment conversation logic internally.
That way you don't force the user to use Moment

Reply:
done